stdlib/REPL/src/REPL.jl

```diff
@@ -774,9 +774,35 @@ find_hist_file() = get(ENV, "JULIA_HISTORY",
 
 backend(r::AbstractREPL) = r.backendref
 
+# Allows an external package to add hooks into the code loading
+# return true if installed, false if not
+# to e.g. install packages on demand
+const install_package_hooks = Any[]
+
 function eval_with_backend(ast, backend::REPLBackendRef)
     put!(backend.repl_channel, (ast, 1))
-    return take!(backend.response_channel) # (val, iserr)
+    resp = take!(backend.response_channel) # (val, iserr)
+    if last(resp)
+        err = first(first(first(resp)))
+        if err isa ModuleNotFoundError
+            handle_install_package_hooks(err.mod, ast, backend)
+        end
+    end
+    return resp
+end
+
+function handle_install_package_hooks(mod, ast, backend)
+    line = string(ast.args[2])
+    # only exercise hooks for lines starting with `using/import` to avoid re-running non-load code
+    if startswith(line, "using") || startswith(line, "import")
```

Comment:
I am not sure this is the best way to check this, since there could also be leading whitespace or the `using` statement could be further nested but still run first. We could do some analysis on the parsed AST, but I am wondering if in practice just always reevaluating the whole input would really be a problem.

Reply:
>but I am wondering if in practice just always reevaluating the whole input would really be a problem.

Yeah, I think you might be right. I've removed the check for using/import